sys/include/net/ipv6/addr.h

```diff
@@ -278,8 +305,9 @@ static inline bool ipv6_addr_is_loopback(const ipv6_addr_t *addr)
  */
 static inline bool ipv6_addr_is_ipv4_compat(const ipv6_addr_t *addr)
 {
-    return (addr->u64[0].u64 == 0) &&
-           (addr->u32[2].u32 == 0);
+    const ipv6_addr_t unspecified = IPV6_ADDR_UNSPECIFIED;
```

Comment:
Correction... no significant change on Arch, but the described change on Ubuntu 14.04